src/app/fdctl/config.c

```diff
@@ -712,207 +733,245 @@ config_parse( int *    pargc,
+        tile->quic.ip_addr = config->tiles.net.ip_addr;
+        fd_memcpy( tile->quic.src_mac_addr, config->tiles.net.mac_addr, 6 );
+        tile->quic.quic_transaction_listen_port = config->tiles.quic.quic_transaction_listen_port;
+        tile->quic.legacy_transaction_listen_port = config->tiles.quic.regular_transaction_listen_port;
+        tile->quic.idle_timeout_millis = config->tiles.quic.idle_timeout_millis;
         break;
       case FD_TOPO_TILE_KIND_VERIFY:
         break;
       case FD_TOPO_TILE_KIND_DEDUP:
-        tile->dedup.tcache_depth = result.tiles.dedup.signature_cache_size;
+        tile->dedup.tcache_depth = config->tiles.dedup.signature_cache_size;
         break;
       case FD_TOPO_TILE_KIND_PACK:
-        tile->pack.max_pending_transactions = result.tiles.pack.max_pending_transactions;
-        tile->pack.bank_tile_count = result.layout.bank_tile_count;
-        strncpy( tile->pack.identity_key_path, result.consensus.identity_path, sizeof(tile->pack.identity_key_path) );
+        tile->pack.max_pending_transactions = config->tiles.pack.max_pending_transactions;
+        tile->pack.bank_tile_count = config->layout.bank_tile_count;
+        strncpy( tile->pack.identity_key_path, config->consensus.identity_path, sizeof(tile->pack.identity_key_path) );
         break;
       case FD_TOPO_TILE_KIND_BANK:
         break;
       case FD_TOPO_TILE_KIND_SHRED:
-        tile->shred.depth = result.topo.links[ tile->out_link_id_primary ].depth;
-        tile->shred.ip_addr = result.tiles.net.ip_addr;
-        fd_memcpy( tile->shred.src_mac_addr, result.tiles.net.mac_addr, 6 );
-        tile->shred.fec_resolver_depth = result.tiles.shred.max_pending_shred_sets;
-        strncpy( tile->shred.identity_key_path, result.consensus.identity_path, sizeof(tile->shred.identity_key_path) );
-        tile->shred.expected_shred_version = result.consensus.expected_shred_version;
-        tile->shred.shred_listen_port = result.tiles.shred.shred_listen_port;
+        tile->shred.depth = config->topo.links[ tile->out_link_id_primary ].depth;
+        tile->shred.ip_addr = config->tiles.net.ip_addr;
+        fd_memcpy( tile->shred.src_mac_addr, config->tiles.net.mac_addr, 6 );
+        tile->shred.fec_resolver_depth = config->tiles.shred.max_pending_shred_sets;
+        strncpy( tile->shred.identity_key_path, config->consensus.identity_path, sizeof(tile->shred.identity_key_path) );
+        tile->shred.expected_shred_version = config->consensus.expected_shred_version;
+        tile->shred.shred_listen_port = config->tiles.shred.shred_listen_port;
         break;
       case FD_TOPO_TILE_KIND_STORE:
         break;
       default:
-        FD_LOG_ERR(( "unknown tile kind %lu", tile->kind ));
+        FD_LOG_RAW_ERR(( "unknown tile kind %lu", tile->kind ));
     }
   }
+}
+
+int
+config_write_memfd( config_t * config ) {
+  int config_memfd = memfd_create( "config", 0 );
```

Comment:
I know this doesn't have to be unique, but I think making it clear its firedancer's config (as opposed to the Labs part maybe) can't hurt, e.g. `fd_config` or `firedancer_config`